cmd/kubectl-gadget/utils/trace.go

```diff
@@ -208,65 +213,236 @@ func GenericTraceCommand(
+			Annotations: map[string]string{
+				GADGET_OPERATION: config.Operation,
+			},
+			Labels: map[string]string{
+				TRACE_ID: traceID,
+				/*
+				 * Add all this information here to be able to find the trace thanks
+				 * to them when calling getTraceListFromParameters().
+				 */
+				"GadgetName":    config.GadgetName,
+				"nodeName":      config.NodeName,
+				"namespace":     config.NamespaceName,
+				"podName":       config.PodName,
+				"containerName": config.ContainerName,
+				"outputMode":    config.TraceOutputMode,
+			},
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   config.NodeName,
+			Gadget: config.GadgetName,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     config.NamespaceName,
+				Podname:       config.PodName,
+				ContainerName: config.ContainerName,
+				Labels:        labelsSelector,
+			},
+			RunMode:    "Manual",
+			OutputMode: config.TraceOutputMode,
+		},
+	}
+
+	err := createTraces(trace)
+	if err != nil {
+		return "", err
+	}
+
+	return traceID, nil
+}
+
+// getTraceListFromOptions returns a list of traces corresponding to the given
+// options.
+func getTraceListFromOptions(listTracesOptions metav1.ListOptions) (gadgetv1alpha1.TraceList, error) {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return gadgetv1alpha1.TraceList{}, err
+	}
+
+	var traces gadgetv1alpha1.TraceList
+
+	err = traceRestClient.
+		Get().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Into(&traces)
+	if err != nil {
+		return traces, err
+	}
+
+	return traces, nil
+}
+
+// getTraceListFromID returns an array of pointer to gadgetv1alpha1.Trace
+// corresponding to the given traceID.
+// If no trace corresponds to this ID, error is set.
+func getTraceListFromID(traceID string) (gadgetv1alpha1.TraceList, error) {
 	var listTracesOptions = metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("trace-template-hash=%s", traceID),
+		LabelSelector: fmt.Sprintf("%s=%s", TRACE_ID, traceID),
 		FieldSelector: fields.Everything().String(),
 	}
 
-	// Wait until results are ready and fetch all results
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return traces, fmt.Errorf("Error getting traces from traceID %q: %w", traceID, err)
+	}
+
+	if len(traces.Items) == 0 {
+		return traces, fmt.Errorf("No traces found for traceID %q!", traceID)
+	}
+
+	return traces, nil
+}
+
+// SetTraceOperation set the operation of an existing trace.
+// If trace does not exist or Annotations does not contain GADGET_OPERATION key
+// an error is returned.
+func SetTraceOperation(traceID string, operation string) error {
+	traces, err := getTraceListFromID(traceID)
+	if err != nil {
+		return err
+	}
+
+	for _, trace := range traces.Items {
+		err = updateTraceOperation(&trace, operation)
+		if err != nil {
+			return fmt.Errorf("error updating trace operation for %s: %w", traceID, err)
+		}
+	}
```

Comment:
I don't know if it could really happen but, theoretically, we could potentially have some traces that were updated and some others that don't. How can the user recover from this situation?

Reply:
hmm, it's a difficult one. Perhaps it should not return the first error that it finds but instead try to continue updating the operation for the other traces. An error should be returned if the operation failed in any trace. (We could try to create an error that contains all the error messages together).

Reply:
Yes, if we decide to try anyway to set the annotation on the other traces, I think we should collect all the errors and report them. Otherwise, it will be pointless to continue iterating.